web/src/people.ts

```diff
@@ -760,28 +761,42 @@ export function should_add_guest_user_indicator(user_id: number): boolean {
     return user.is_guest;
 }
 
-export function user_can_direct_message(recipient_ids_string: string): boolean {
-    // Common function for checking if a user can send a direct
-    // message to the target user (or group of users) represented by a
-    // user ids string.
-
-    // Regardless of policy, we allow sending direct messages to bots and to self.
+export function user_can_initiate_direct_message_thread(recipient_ids_string: string): boolean {
+    const direct_message_initiator_group_id = realm.realm_direct_message_initiator_group;
     const recipient_ids = user_ids_string_to_ids_array(recipient_ids_string);
-    if (
-        recipient_ids.length === 1 &&
-        recipient_ids[0] !== undefined &&
-        (is_valid_bot_user(recipient_ids[0]) || is_my_user_id(recipient_ids[0]))
-    ) {
+    let is_recipient_bot_or_self = true;
+    for (const recipient of recipient_ids) {
+        if (!is_valid_bot_user(recipient) && !(recipient === my_user_id)) {
+            is_recipient_bot_or_self = false;
+        }
+    }
+    if (is_recipient_bot_or_self) {
         return true;
     }
+    return is_user_in_group(direct_message_initiator_group_id, my_user_id);
+}
 
-    if (
-        realm.realm_private_message_policy ===
-        settings_config.private_message_policy_values.disabled.code
-    ) {
-        return false;
+export function user_can_direct_message(recipient_ids_string: string): boolean {
+    const direct_message_permission_group_id = realm.realm_direct_message_permission_group;
+    const recipient_ids = user_ids_string_to_ids_array(recipient_ids_string);
+    let is_recipient_bot_or_self = true;
```

Comment:
This name is confusing. Try `other_human_recipients_exist` with the opposite meaning.